cmd/gmailctl/cmd/init_cmd.go

```diff
@@ -123,14 +117,13 @@ func createDefault(path, contents string) (err error) {
 	return nil
 }
 
-func resetLib() (err error) {
+func updateLib() (err error) {
 	libFile := path.Join(cfgDir, "gmailctl.libsonnet")
-	libFileOld := path.Join(cfgDir, "gmailctl.libsonnet.backup")
 
 	// check for existence of current libFile
 	if _, err := os.Stat(libFile); err == nil {
```

Comment:
I think you can simplify all the logic here with some refactoring:

* [handleCfgDir](https://cs.github.com/mbrt/gmailctl/blob/08c9eee2d0535ebd3cc64e614249eb58299bba71/cmd/gmailctl/cmd/init_cmd.go#L75) will create the directory for you if it's not present.
* [createDefault](https://cs.github.com/mbrt/gmailctl/blob/08c9eee2d0535ebd3cc64e614249eb58299bba71/cmd/gmailctl/cmd/init_cmd.go#L95) creates the files if it's not there already.

Both functions do almost what you want, but not exactly, so you need to extract some functionality. Something like this should work:

```
if err := handleCfgDir(); err != nil {
    return err
}
return createFile(libFile, data.GmailctlLib())
```

And `createFile` would just be similar to `createDefault` but without the initial check for existence:

```
func createDefault(path, contents string) (err error) {
	f, err := os.Create(path)
	if err != nil {
		return err
	}
	defer func() {
		e := f.Close()
		if err == nil {
			err = e
		}
	}()
	_, err = f.WriteString(contents)
	return err
}
```

Reply:
Thanks again for the pointers - this helped me a lot in reading through and better understanding the code